### PR TITLE
balloon_check: Add new case to test option free-page-reporting

### PR DESCRIPTION
**Pull request by huangyum**
Signed-off-by: Yumei Huang <yuhuang@redhat.com>
Depends on https://github.com/avocado-framework/avocado-vt/pull/2978
ID: 1944030

### CODE REVIEW

**Comment by huangyum** [comment]
(1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.balloon_check.balloon_evict_and_enlarge.balloon_free: PASS (154.90 s)
(2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.balloon_check.balloon_evict_and_enlarge.balloon_free: PASS (389.94 s)
@yanan-fu @MiriamDeng  Pls help review, thanks.

File: qemu/tests/balloon_check.py [hunk 1]
```diff
@@ -614,5 +616,20 @@ def run(test, params, env):
             return
     try:
         balloon_test.reset_memory()
+        if params.get("balloon_opt_free_page_reporting") == "yes" and \
+                balloon_test.params.get("os_type") == "linux":
```
Comment (yanan-fu) [inline qemu/tests/balloon_check.py:620]: we can  use `params['os_type']` directly.

File: qemu/tests/balloon_check.py [hunk 2]
```diff
@@ -614,5 +616,20 @@ def run(test, params, env):
...
+            get_res_cmd = params["get_res_cmd"] % balloon_test.vm.get_pid()
+            memhog_cmd = params["memhog_cmd"]
+            consumed_mem = float(normalize_data_size(params["consumed_mem"]))
+            res1 = float(normalize_data_size(process.getoutput(get_res_cmd)))
+            session = balloon_test.vm.wait_for_login()
+            session.cmd_output_safe(memhog_cmd)
+            res2 = float(normalize_data_size(process.getoutput(get_res_cmd)))
+            time.sleep(30)
```
Comment (MiriamDeng) [inline qemu/tests/balloon_check.py:628]: I remember that it's not preferable time.sleep (xx)

File: qemu/tests/balloon_check.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/balloon_check.py]: Thanks, updated.

File: qemu/tests/balloon_check.py (hunk 2 above)
Comment (huangyum) [reply qemu/tests/balloon_check.py:628]: The step is to check if memory usage returns back after 30s. Do you have better idea? I think it's not necessary to use wait_for() and check res again and again in 30s. What do you think?

**Comment by MiriamDeng** [comment]
On power,
avocado.core.exceptions.TestFail: QEMU should consume same memory as before memhog
 (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.balloon_check.balloon_evict_and_enlarge.balloon_free: FAIL: QEMU should consume same memory as before memhog  (197.01 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by huangyum** [comment]
> On power,
> avocado.core.exceptions.TestFail: QEMU should consume same memory as before memhog
> (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.balloon_check.balloon_evict_and_enlarge.balloon_free: FAIL: QEMU should consume same memory as before memhog (197.01 s)
> RESULTS : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

Would you please help check if power kernel supports this feature?  Thanks.

**Comment by MiriamDeng** [comment]
I can run it pass by manual.
Guest memory is 8G 
memhog -r10 4G

File: qemu/tests/balloon_check.py [hunk 3]
```diff
@@ -614,5 +616,22 @@ def run(test, params, env):
...
+            res3 = float(normalize_data_size(process.getoutput(get_res_cmd)))
+            logging.info("The RES values are %sM, %sM, and %sM sequentially",
+                         res1, res2, res3)
+            if res2 - res1 < consumed_mem * 0.5:
+                test.error("QEMU should consume more memory")
+            if res3 - res1 > res1 * 0.1:
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/balloon_check.py:634]: Sure, please tell the proper consumed_mem, sleep time, and compare ratio for ppc, thanks.

[the remark this answers is not in the document]
Comment (MiriamDeng) [reply qemu/tests/balloon_check.py:634]: Can we set it to 4G, sleep time is acceptable, I think
#memhog -r10 4G

[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/balloon_check.py:634]: In that case, we need to make sure guest has more than 4G free memory. We have some x86 hosts with small memory. How about set 4G only for ppc?

[the remark this answers is not in the document]
Comment (MiriamDeng) [reply qemu/tests/balloon_check.py:634]: Ok, sure

**Comment by huangyum** [comment]
@MiriamDeng Updated as we discussed, pls help review again, thanks!

File: qemu/tests/cfg/balloon_check.cfg [hunk 4]
```diff
@@ -17,6 +17,16 @@
...
             balloon_opt_deflate_on_oom = yes
+        - balloon_free:
+            only balloon_evict_and_enlarge
+            required_qemu = [5.2.0, )
+            no RHEL.8.3 RHEL.8.2 RHEL.8.1 RHEL.8.0 RHEL.7
+            balloon_opt_free_page_reporting = yes
+            get_res_cmd = "top -n1 -b -p %s | grep qemu-kvm | awk -F ' ' '{print $6}'"
+            consumed_mem = 2G
+            ppc64,ppc64le:
+                onsumed_mem = 4G
```
Comment (yanan-fu) [inline qemu/tests/cfg/balloon_check.cfg:28]: Typo, `consumed_mem`

Comment (huangyum) [reply qemu/tests/cfg/balloon_check.cfg]: Oh, it must be some paste error. Updated, thanks.

**Comment by MiriamDeng** [comment]
LGTM
06:30:11 INFO | PASS 1-Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.balloon_check.balloon_evict_and_enlarge.balloon_free

**Comment by yanan-fu** [comment]
Thanks all!

**Pull request merged**